core/src/main/scala/kafka/raft/KafkaMetadataLog.scala

```diff
@@ -674,7 +675,7 @@ object KafkaMetadataLog extends Logging {
     logDir: Path,
     expiredSnapshots: mutable.TreeMap[OffsetAndEpoch, Option[FileRawSnapshotReader]],
     logging: Logging
-  ): () => Unit = () => {
+  ): Unit = {
```

Comment:
This made the code less clear, it's better to have the clarity that a lambda is being passed.